<launch>
  <node pkg="plane_detector" type="detect_plane" name="detect_plane" output="screen">
    <param name="board_num" type="int" value="23"/>
    <param name="write_all" type="bool" value="true"/>
    <param name="write_all_path" type="string" value="$(find plane_detector)/data/filter/init_l.json"/>
    <param name="pointcloud_path" type="string" value="$(find plane_detector)/data/calib/hkust4/"/>
    <param name="write_path" type="string" value="$(find plane_detector)/data/filter/hkust4/"/>
    <param name="boundary_param" type="string" value="$(find plane_detector)/config/hkust4/"/>
    <param name="max_SVD_iteration" type="int" value="30"/>
    <param name="inlier_SVD_ratio" type="double" value="0.95"/>
    <param name="max_SVD_value" type="double" value="0.9"/>
    <param name="max_circle_radius" type="double" value="0.2"/>
  </node>

  <arg name="rviz" default="true"/>

  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find plane_detector)/rviz/rosrun.rviz"/>
  </group>
</launch>